### PR TITLE
[NFC][VPlan] Simplify VPValue::removeUser

### PR DESCRIPTION
**Pull request by sunshaoce**
I think we don't need to use lambda, and we can end the loop in advance like this.

### CODE REVIEW

File: llvm/lib/Transforms/Vectorize/VPlanValue.h [hunk 1]
```diff
@@ -121,18 +121,14 @@ class VPValue {
...
-    bool Found = false;
     // The same user can be added multiple times, e.g. because the same VPValue
     // is used twice by the same VPUser. Remove a single one.
-    erase_if(Users, [&User, &Found](VPUser *Other) {
-      if (Found)
-        return false;
-      if (Other == &User) {
-        Found = true;
-        return true;
+    for (const auto &U : Users) {
```
Comment (topperc) [inline llvm/lib/Transforms/Vectorize/VPlanValue.h:126]: Could we do

```
auto I = llvm::find(Users, User);
if (I != Users.end())
  Users.erase(I);
```

Or its expected to always find it, you could do
`Users.erase(llvm::find(Users, User))`

**Review by topperc — APPROVED** [review]
LGTM

File: llvm/lib/Transforms/Vectorize/VPlanValue.h [hunk 2]
```diff
@@ -121,18 +121,11 @@ class VPValue {
...
-    erase_if(Users, [&User, &Found](VPUser *Other) {
-      if (Found)
-        return false;
-      if (Other == &User) {
-        Found = true;
-        return true;
-      }
-      return false;
-    });
+    auto *I = llvm::find(Users, &User);
```
Comment (fhahn) [inline llvm/lib/Transforms/Vectorize/VPlanValue.h:126]: nit: `llvm::` shouldn't be necessary.

**Review by fhahn — APPROVED** [review]
LGTM, thanks. 

Please see the inline comment before merging and also adjust the commit messages to be definitive, i.e. drop `It think` and adjust to say hat this is now using find + erase; there's no more explicit loop.

**Pull request merged**